### ISSUE TITLE
Consider query-only test time training

### ISSUE THREAD

**Issue by mseeger** [awslabs/keys_values#132]
**Is your feature request related to a problem? Please describe.**
Look at Bansal etal, ICLR 2026: "Test-time training for long-context LLMs". How difficult is this to implement?

**Comment by mseeger** [comment]
The idea is to first run prefilling on the prompt and then run a very efficient form of training on this data, using the usual next-token prediction loss. Efficiency is obtained by a few restrictions:
* All key and value vectors are frozen after the prefilling. They are taken out of the computation graph and treated as constant. This
   means that in the multi-head attention blocks, only the query vectors are updated based on weights and inputs.
* Only the weights of query linear maps are updated. All other model weights remain frozen.
* Context is limited. They suggest to randomly draw subsequences from the prompt of a length much shorter than the prompt. Tokens
  within a subsequence are predicted from earlier tokens of the subsequence. It is not clear from the paper whether they (1) simply
  start keys and values with the subsequence, or (2) use fixed keys and values from before the subsequence as well. The latter
  would be more realistic, but also more expensive.

This is a neat paper, and seems a good fit to be combined with sparse attention and our techniques:
* Training with fixed, constant keys and values could neatly be implemented in our codecase. We just abstract the computation of
   keys and values from inputs and weights in `model.py` in some method. In a variant, these would be loaded from the KV cache
   instead.
* They need gradients. Maybe our methods can be helpful?
* Say a subsequence starts at token position $t$. An important question is what keys and values from before $t$ are to be used in the
   loss and gradient computation. Extreme cases are (1) use no keys, values before $t$ and (2) use all keys, values before $t$. For
   sparse attention, the latter means we use the full KV cache. If only a part of the KV cache is used, how is it selected? We could use
   the H2O score for that. In order to use 10\% of the cache, we just "evict" 90\% virtually for computing the loss.
* They draw subsequences from the prompt at random. But is this what we want? The prompt will have structure, such as user-agent
   separation, or different types of content (e.g., code versus text). Maybe we should only train to predict tokens which came from the
   agent earlier on? In other words, the TTT should be steered towards alignment rather than pretraining.
